pandas/io/excel/_base.py

```diff
@@ -902,6 +913,12 @@ def __init__(
         self, path_or_buffer, engine=None, storage_options: StorageOptions = None
     ):
         if engine is None:
+            warnings.warn(
```

Comment:
Shouldn't we be more specific about *when* to warn?

* If people only have openpyxl installed, shouldn't we default to use that instead of having the warning?
* From earlier discussion, I understood that we are (for now) fine to keep xlrd support for old excel files for which it is the only. So for those cases (eg `.xls`), we shouldn't raise a warning?

Reply:
and python < 3.9

Reply:
@jorisvandenbossche I've implemented the xls check, however I don't understand your first bullet. Isn't having the deprecation warning so that we don't change behavior without notice? Why is it okay to change behavior if they have openpyxl installed?

Reply:
> Why is it okay to change behavior if they have openpyxl installed?

My suggestions was for when they have *only* openpyxl installed. It will currently not be a very common situation, since we don't try to use openpyxl if xlrd is not present, unless the user actually specifies `engine="openpyxl"`. But, given the deprecation warnings, this could be another solution for users: ensure you update your environment to only have openpyxl and not xlrd, and then you don't need to update all occurrences of `read_excel` in your code.

Reply:
Ahh, thanks! I completely misread.